### ISSUE TITLE
The g:mirror#config_path setting does not accept '~', requires absolute path

### ISSUE THREAD

**Issue by jakubgs** [zenbro/mirror.vim#5]
I just started playing with `mirror.vim` and I really like it but I wasted like 10 minutes trying to figure out why my configuration was not being loaded. And that was because my config looked like this:

```
let g:mirror#config_file = '~/.config/nvim/mirrors.yml'
```

Only after I changed that to an absolute path it started to work. Just a small nitpick that might annoy new users.

**Comment by zenbro** [comment]
Hi!
Relative path is already supported.
Looks like an error in option name:
`let g:mirror#config_file = '~/.config/nvim/mirrors.yml'`
Correct way:
`let g:mirror#config_path = '~/.config/nvim/mirrors.yml'`

**Comment by jakubgs** [comment]
Oh, sorry, that was my mistake since I typed that out. I just tested this again, this is an exact copy of what I have in my `init.vim`:

``` vim
let g:mirror#config_path = '~/.config/nvim/mirrors.yml'
"let g:mirror#config_path = '/home/local/ANT/<omitted_username>/.config/nvim/mirrors.yml'
```

This way it doesn't work. I try `MirrorPush` in the configured path and:

```
E492: Not and editor command: MirrorPush
```

If I uncomment the second line it works.

When I comment out both and I run `let g:mirror#config_path` I see this:

```
g:mirror#config_path    /home/local/ANT/<omitted_username>/.mirrors
```

So the default value is using the absolute path.

PERHAPS this is due to the fact that my home path is a bit unusual. Not sure. I'll do some more tests tomorrow, for now I'm going home(this is all in my work setup).

**Comment by zenbro** [comment]
Can you try to replace `~` with `'$HOME' .`? 
Like this:

```
let g:mirror#config_path = $HOME . '/.config/nvim/mirrors.yml'
```

**Comment by jakubgs** [comment]
Yup, that works just fine.

**Comment by jakubgs** [comment]
But then again I might as well do this since it works too:

``` vim
let g:mirror#config_path = expand('~/.config/nvim/mirrors.yml')
```

**Comment by zenbro** [comment]
I added `expand` for config and cache paths.
Hope it helps.

**Comment by jakubgs** [comment]
Well, I see your change:

```
let g:mirror#config_path = expand(get(g:, 'mirror#config_path', '~/.mirrors'))
```

But as far as I can tell if I set `g:mirror#config_path` in my `init.vim` then it is set AFTER the plugin is loaded(I'm using [vim-plug](https://github.com/junegunn/vim-plug)) so it still results in it being set to exactly what I set.

That's annoying, but I guess it makes sense, your vim config should override anything and everything so it's loaded last.

**Comment by zenbro** [comment]
That sad.
At least you can use expand in your init.vim.

**Comment by jakubgs** [comment]
I just checked how this is done in another plugin I use and one that respects `~` in a path and I found this for the `g:easytags_by_filetype` config:

https://github.com/xolox/vim-easytags/blob/master/autoload/xolox/easytags.vim#L189

He's using his own `xolox#misc#path#absolute()` function every time he uses the configured path.

I guess it's fine if it stays the way it is but I would point this out in the documentation to avoid people trouble. Thanks for help, you are awesome!
